network-api/networkapi/wagtailpages/tests/research_hub/test_detail_page.py

```diff
@@ -33,6 +36,60 @@ def test_research_detail_page_breadcrumbs(self) -> None:
         self.assertEqual(len(breadcrumbs), 2)
         self.assertEqual(breadcrumbs, expected_breadcrumbs)
 
+    def test_get_research_authors(self) -> None:
+        """
+        This method should return the profiles of all related research authors.
+        """
+        author_profiles = []
+        detail_page = detail_page_factory.ResearchDetailPageFactory(parent=self.library_page, research_authors=[])
+
+        for _ in range(3):
+            research_author_relation = relations_factory.ResearchAuthorRelationFactory(
+                research_detail_page=detail_page
+            )
+            author_profiles.append(research_author_relation.author_profile)
+
+        research_authors = detail_page.get_research_authors()
+
+        self.assertEqual(len(research_authors), 3)
+        self.assertCountEqual(author_profiles, research_authors)
```

Comment:
I think it would be best to assert that the author profiles actually show up in `research_authors` instead of counts, something like:

```python
self.assertIn(author_profiles[0], research_authors)
self.assertIn(author_profiles[1], research_authors)
self.assertIn(author_profiles[2], research_authors)
```

And if we had a profile that was not supposed to be in `research_authors`:

```python
self.assertNotIn(profile_not_in_this_page, research_authors)
```

Reply:
Thanks for the suggestion Jhonatan!

[AssertCountEqual](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual) actually does not check the count of items, but instead checks that two items contain the same elements, regardless of their order. 

However, after your suggestion I can see that the name is very misleading, so much so that there has been a [github issue opened over it](https://github.com/python/cpython/issues/71258)! Though in the ticket they decided to leave the name as-is.

I like your suggestion of checking for each item with their index instead, as its easier to read/understand and have added it to my PR 👍 

I also added the other suggestion of creating an unrelated page and author and double checking that they do not appear in the list.